dataprofiler/profilers/graph_profiler.py

```diff
@@ -0,0 +1,339 @@
+                        best_mle = (distribution.name, mle, pdf, ktest)
+                continuous_distributions[attribute] = best_mle
+            else:
+                continuous_distributions[attribute] = None
+
+        return continuous_distributions
+
+    @staticmethod
+    def _get_categorical_distribution(self, graph):
+        """
+        Compute categorical probabilities of graph edge categorical attributes
+        """
+        attributes = GraphProfile.find_all_attributes(graph)
+        categorical_attributes = GraphProfile._get_categorical_attributes(graph)
+
+        distribution_candidates = [st.bernoulli, st.binom, st.geom, st.poisson]
+        categorical_distributions = dict()
+
+        for attribute in attributes:
+            if attribute in categorical_attributes:
+                best_mle = ("initial value", 1000)
+                data_as_list = self._attribute_data_as_list(graph, attribute)
+
+                for distribution in distribution_candidates:
+                    # compute fit, mle, kolmogorov-smirnov test to test fit, and pdf
+                    fit = distribution.fit(data_as_list)
+                    mle = distribution.nnlf(fit, data_as_list)
+                    ktest = st.kstest(data_as_list, "norm", fit)
+
+                    test_points = np.linspace(
+                        min(data_as_list), max(data_as_list), 1000
+                    )
+                    pdf = distribution.pdf(test_points, fit)
+
+                    if mle <= best_mle[0]:
+                        best_mle = (distribution.name, mle, pdf, ktest)
+                categorical_distributions[attribute] = best_mle
+            else:
+                categorical_distributions[attribute] = None
+
+        return categorical_distributions
+
+    @staticmethod
+    def _get_categorical_and_continuous_attributes(graph):
+        """
+        Finds and lists categorical and continuous attributes
+        """
+        categorical_attributes = []
+        continuous_attributes = []
+        attributes = GraphProfile._find_all_attributes(graph)
+        for attribute in attributes:
+            is_categorical = False
+            for u, v in graph.edges():
+                attribute_value = graph[u][v][attribute]
+                if float(attribute_value).is_integer():
+                    is_categorical = True
+                    break
+            if is_categorical:
+                categorical_attributes.append(attribute)
+            else:
+                continuous_attributes.append(attribute)
+        return (categorical_attributes, continuous_attributes)
+
+    """
+    Helper functions
+    """
+
+    @staticmethod
+    def _find_all_attributes(graph):
+        """
+        Compute the number of attributes for each edge
+        """
+        attribute_list = set(
+            np.array([list(graph.edges[n].keys()) for n in graph.edges()]).flatten()
+        )
+        return list(attribute_list)
```

Comment:
do we have to search the graph for this? could this be provided by the reader? just thinking outloud.